tests/data/diagnosis_summary.md

```diff
@@ -0,0 +1,4 @@
+| machine | Category | Defective Details | kernel-launch/event_overhead:0 | kernel-launch/event_overhead:1 | kernel-launch/event_overhead:2 | kernel-launch/event_overhead:3 | kernel-launch/event_overhead:4 | kernel-launch/event_overhead:5 | kernel-launch/event_overhead:6 | kernel-launch/event_overhead:7 | kernel-launch/return_code | kernel-launch/wall_overhead:0 | kernel-launch/wall_overhead:1 | kernel-launch/wall_overhead:2 | kernel-launch/wall_overhead:3 | kernel-launch/wall_overhead:4 | kernel-launch/wall_overhead:5 | kernel-launch/wall_overhead:6 | kernel-launch/wall_overhead:7 | mem-bw/D2H_Mem_BW:0 | mem-bw/D2H_Mem_BW:1 | mem-bw/D2H_Mem_BW:2 | mem-bw/D2H_Mem_BW:3 | mem-bw/D2H_Mem_BW:4 | mem-bw/D2H_Mem_BW:5 | mem-bw/D2H_Mem_BW:6 | mem-bw/D2H_Mem_BW:7 | mem-bw/H2D_Mem_BW:0 | mem-bw/H2D_Mem_BW:1 | mem-bw/H2D_Mem_BW:2 | mem-bw/H2D_Mem_BW:3 | mem-bw/H2D_Mem_BW:4 | mem-bw/H2D_Mem_BW:5 | mem-bw/H2D_Mem_BW:6 | mem-bw/H2D_Mem_BW:7 | mem-bw/return_code |
+| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
+| sb-validation-01 | KernelLaunch | kernel-launch/event_overhead:0(B/L: 0.0060 VAL: 0.1000 VAR: 1577.85% Rule:lambda x:x>0.05) | 15.778523489932889 | -0.0016778523489932202 | -0.0654362416107382 | -0.07718120805369133 | -0.0067114093959731716 | -0.011744966442952979 | -0.04026845637583889 | -0.010067114093959759 | 0.0 | 0.0 | 0.0 | 0.01949317738791428 | 0.022417153996101297 | 0.03606237816764138 | -0.01949317738791428 | 0.01851851851851844 | 0.04385964912280709 | 0.0 | 0.012345679012345708 | 0.00823045267489709 | 0.012345679012345708 | 0.0 | 0.0 | -0.016460905349794327 | 0.012345679012345708 | 0.0 | 0.007812499999999972 | 0.015624999999999944 | 0.01953125 | 0.023437499999999917 | 0.007812499999999972 | -0.011718750000000028 | 0.01953125 | 0.0 |
+| sb-validation-03 | FailedTest,Mem | mem-bw/D2H_Mem_BW:0_miss,mem-bw/D2H_Mem_BW:1_miss,mem-bw/D2H_Mem_BW:2_miss,mem-bw/D2H_Mem_BW:3_miss,mem-bw/D2H_Mem_BW:4_miss,mem-bw/D2H_Mem_BW:5_miss,mem-bw/D2H_Mem_BW:6_miss,mem-bw/D2H_Mem_BW:7_miss,mem-bw/H2D_Mem_BW:0_miss,mem-bw/H2D_Mem_BW:1_miss,mem-bw/H2D_Mem_BW:2_miss,mem-bw/H2D_Mem_BW:3_miss,mem-bw/H2D_Mem_BW:4_miss,mem-bw/H2D_Mem_BW:5_miss,mem-bw/H2D_Mem_BW:6_miss,mem-bw/H2D_Mem_BW:7_miss,mem-bw/return_code(VAL: 1.0000 Rule:lambda x:x>0) | 0.0 | -0.0016778523489932202 | -0.0654362416107382 | -0.07718120805369133 | -0.0067114093959731716 | -0.011744966442952979 | -0.04026845637583889 | -0.010067114093959759 | 0.0 | 0.0 | 0.0 | 0.01949317738791428 | 0.022417153996101297 | 0.03606237816764138 | -0.01949317738791428 | 0.01851851851851844 | 0.04385964912280709 | nan | nan | nan | nan | nan | nan | nan | nan | nan | nan | nan | nan | nan | nan | nan | nan | 1.0 |
```

Comment:
For the output, can we decide how to round numbers to a given number of decimal places? It's really strange for -0.00167xxxxxxx for markdown.

Reply:
sure, added a new optional arg in run() to round the decimal digits